spring-vault-core/src/main/java/org/springframework/vault/support/VaultResponseDataVersion2.java

```diff
@@ -0,0 +1,34 @@
+package org.springframework.vault.support;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import java.util.Map;
+import org.springframework.lang.Nullable;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class VaultResponseDataVersion2<T> {
```

Comment:
Any reason why we require an additional response wrapper and why we cannot follow the imperative typing approach?

Reply:
My goal was to force WebClient to do all JSON marshaling. My intent was to use any added Jackson deserialization to the DI'ed ` WebClient.Builder`.

To do what imperative did - do you know of a way to get the default Jackson deserialization configuration from WebClient?